core/domain/stats_services_test.py

```diff
@@ -648,51 +768,66 @@ def test_get_stats_for_new_exp_version(self):
+            list(exploration.states.keys()),
+            exp_versions_diff,
+            None
+        )
         stats_services.create_stats_model(exploration_stats)
 
-        exploration_stats = stats_services.get_exploration_stats_by_id(
+        exp_stats = stats_services.get_exploration_stats_by_id(
             exploration.id, exploration.version)
-        self.assertEqual(exploration_stats.exp_version, 7)
+        # Ruling out the possibility of None for mypy type checking.
+        assert exp_stats is not None
+        self.assertEqual(exp_stats.exp_version, 7)
         self.assertEqual(
-            set(exploration_stats.state_stats_mapping.keys()),
+            set(exp_stats.state_stats_mapping.keys()),
             set(['Home', 'New state 4', 'Renamed state', 'End']))
 
         # Test the values of the stats carried over from the last version.
-        self.assertEqual(exploration_stats.num_actual_starts_v2, 5)
-        self.assertEqual(exploration_stats.num_completions_v2, 2)
+        self.assertEqual(exp_stats.num_actual_starts_v2, 5)
+        self.assertEqual(exp_stats.num_completions_v2, 2)
         self.assertEqual(
-            exploration_stats.state_stats_mapping['Home'].total_hit_count_v2, 8)
+            exp_stats.state_stats_mapping['Home'].total_hit_count_v2, 8)
         self.assertEqual(
-            exploration_stats.state_stats_mapping[
+            exp_stats.state_stats_mapping[
                 'Renamed state'].first_hit_count_v2, 2)
         self.assertEqual(
-            exploration_stats.state_stats_mapping[
+            exp_stats.state_stats_mapping[
                 'End'].useful_feedback_count_v2, 4)
         # State 'New state 4' has been deleted and recreated, so it should
         # now contain default values for stats instead of the values it
         # contained in the last version.
         self.assertEqual(
-            exploration_stats.state_stats_mapping[
+            exp_stats.state_stats_mapping[
                 'New state 4'].total_answers_count_v2, 0)
 
         # Test reverts.
         exploration.version += 1
         exploration_stats = stats_services.get_stats_for_new_exp_version(
-            exploration.id, exploration.version, exploration.states,
-            None, 5)
+            exploration.id,
+            exploration.version,
+            list(exploration.states.keys()),
+            None,
+            5
+        )
         stats_services.create_stats_model(exploration_stats)
 
-        exploration_stats = stats_services.get_exploration_stats_by_id(
-            exploration.id, exploration.version)
-        self.assertEqual(exploration_stats.exp_version, 8)
+        exploration_stats_with_none = (
+            stats_services.get_exploration_stats_by_id(
+                exploration.id, exploration.version
+            )
+        )
+        # Ruling out the possibility of None for mypy type checking.
+        assert exploration_stats_with_none is not None
```

Comment:
Same thing re the name, ditto below. I think if it's just a single item (not a list) and you assert it's not None immediately after, then you don't need the "with none" in the name.

Reply:
Same as above, name is changed to `exploration_stats_domain_object`

Reply:
See my concern above with this, it applies here too.

In general don't use the suffix _domain_object.

Reply:
Name changed PTAL!